Orange/widgets/unsupervised/owtsne.py

```diff
@@ -254,12 +254,16 @@ def commit(self):
         self.send_preprocessor()
 
     def _get_projection_data(self):
-        if self.data is None or self.projection is None:
+        if self.data is None:
             return None
+        if self.projection is None:
+            variables = self._get_projection_variables()
+        else:
+            variables = self.projection.domain.attributes
         data = self.data.transform(
             Domain(self.data.domain.attributes,
                    self.data.domain.class_vars,
-                   self.data.domain.metas + self.projection.domain.attributes))
+                   self.data.domain.metas + variables))
```

Comment:
So if I got this right, you're just adding the two t-SNE variables with empty values until the embedding is actually computed and populated with the actual embedding values. That seems all right.

Ideally, you wouldn't even need to draw anything until the embedding finishes, but I don't know how hard that may be to implement...